 question  ,  "executorservice;  interruptedexception;  public  class  ittest  {static  class  sth  {public  void  useless()  throws  interruptedexception  {thread.sleep(3000);}}static  class  runner  implements  runnable  {sth  f;public  runner(sth  f)  {super();this.f  =  f;}@overridepublic  void  run()  {try  {f.useless();}  catch  (interruptedexception  e)  {system.out.println(""it!"");}}}public  static  void  main(string[]  args)  throws  interruptedexception'  '  executionexception  {executorservice  es  =  executors.newcachedthreadpool();sth  f  =  new  sth();future<?>  lo  =  es.submit(new  runner(f));lo.cancel(true);  es.shutdown();}"  ,if  i  have  an  to  which  i  feed  runnabl  task  can  i  select  one  and  interrupt  it  i  know  i  can  cancel  the  futur  return  also  mention  but  how  can  i  rais  an  cancel  doesnt  seem  to  do  it  event  though  it  should  by  look  at  the  sourc  mayb  the  osx  implement  differ  at  least  this  snippet  doesnt  print  it  mayb  im  misunderstan  someth  and  it  not  the  custom  runnabl  that  get  the  except  , 